Add initial codegen documentation (#696)
snowplow/documentation: docs/collecting-data/code-generation/snowtype-config/index.md

---
title: "Snowtype configuration options"
sidebar_position: 3
---

The Snowtype CLI configuration will be saved in a`snowtype.config.json` file after initialization. **We highly recommend you keep this file in the root of your project folder.**

## Attributes in `snowtype.config.json`

### `igluCentralSchemas`

> The schema tracking URLs from schemas available in [Iglu Central](http://iglucentral.com/).

Reviewer: ```suggestion
The URLs for schemas available in [Iglu Central](http://iglucentral.com/).
```

Author: Would keep the `schema tracking URLs` since it is the title of the value we need.

Reviewer: where's that title used? I haven't seen it before. It feels hard to parse. Like, what is a `tracking URL`? Does this sentence mean "the schema that tracks URLs from schemas in Iglu Central"??
I would just call them `schema URIs`

Author: <img width="560" alt="image" src="https://github.com/snowplow/documentation/assets/15251081/8d694412-9418-44fa-b811-d3afd3381c2d">

Reviewer: Oh, in that case, let's add a labelled screenshot like this to the page

Reviewer: Or put `schema tracking URLs` as inline code

### `dataStructures`

> The schema tracking URLs from Data Structures published in the Console.

Reviewer: ```suggestion
The URLs for Data Structures published in the Console.
```

Author: ^^

### `trackingScenarioIds`

The Tracking Scenario IDs you wish to generate tracking code for. The Tracking Scenario ID is a UUID that can be retrieved as the final part of the URL when visiting a tracking scenario main page.

### `organizationId`

> The Organization ID for your Snowplow BDP account. The Organization ID is a UUID that can be retrieved from the URL immediately following the .com when visiting console.

Reviewer: link to Console?
